### PR TITLE
Do not ship rabbitmq-c with syslog-ng (remove as submodule)

### LINKED ISSUE #2051: make of 3.15.1 fails tools/publish.c, subsequent make runs to completion (publish.c skipped?)
# syslog-ng
## Version of syslog-ng
3.15.1
```
## Platform

Oracle Linux 3.8.13-118.20.3.el7uek.x86_64


# Issue

Configured with:
./configure --enable-manpages --enable-systemd --with-systemdsystemunitdir="/usr/lib/systemd/system"

configure works fine...make fails

## Failure


  CCLD     tools/amqp-delete-queue
  CC       tools/tools_amqp_get-get.o
  CCLD     tools/amqp-get
  CC       tools/tools_amqp_publish-publish.o
tools/publish.c: In function ‘main’:
tools/publish.c:108:22: error: ‘POPT_ARG_ARGV’ undeclared (first use in this function)
       "header", 'H', POPT_ARG_ARGV, &headers, 0,
                      ^
tools/publish.c:108:22: note: each undeclared identifier is reported only once for each function it appears in
make[4]: *** [tools/tools_amqp_publish-publish.o] Error 1
make[3]: *** [all] Error 2
make[2]: *** [modules/afamqp/rabbitmq-c/librabbitmq/librabbitmq.la] Error 2
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

### PR DESCRIPTION
**Pull request by Kokan**
This removes the internal version of rabbitmq-c so its source won't be shipped with syslog-ng anymore.

At this point we are stuck with the current commit for rabbitmq-c, because it only supports `cmake` based build after this commit, the autotools have been dropped. The autotools probably won't be supported anymore, and I do not think that syslog-ng project should continue to support it either. Thus I am proposing to drop as a submodule and everybody that needs amqp just get themself as other dependencies.

Fixes #2051
Disclaimer: although it is not a real fix, I believe this is a way forward, and with this that ticket can be closed.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: .travis.yml [hunk 1]
```diff
@@ -54,7 +55,7 @@ before_script:
       --prefix=$HOME/install/syslog-ng
       --with-ivykis=internal
       --with-mongoc=internal
-      --with-librabbitmq=internal
+      --with-librabbitmq=system
```
Comment (furiel) [inline .travis.yml]: In configure.ac you dropped the whole option, so you can just delete the line.

File: configure.ac [hunk 2]
```diff
@@ -1324,47 +1319,8 @@ dnl ***************************************************************************
...
-        LIBRABBITMQ_CFLAGS="-I\$(top_srcdir)/modules/afamqp/rabbitmq-c/librabbitmq -I\$(top_builddir)/modules/afamqp/rabbitmq-c/librabbitmq"
-    else
-        AC_MSG_WARN([Internal librabbitmq-client sources not found in modules/afamqp/rabbitmq-c])
-        with_librabbitmq_client="no"
-    fi
-elif test "x$with_librabbitmq_client" = "xsystem"; then
-    PKG_CHECK_MODULES(LIBRABBITMQ, librabbitmq >= $LIBRABBITMQ_MIN_VERSION,, enable_amqp="no")
-fi
-
-if test "x$with_librabbitmq_client" = "xno"; then
```
Comment (furiel) [inline configure.ac]: Similar validation still needed: if amqp is enabled, librabbitmq-client must be available.

**Comment by algernon** [comment]
I wholeheartedly support the idea! \o/

**Comment by bazsi** [comment]
Hi,

Apart from ubuntu trusty, all relevant debian/ubuntu versions meet the
librabbitmq_min_version criteria.

Centos6/7 is a question still and how we would address the trusty case in
our obs packages.

@czanik @lbudai

File: .travis.yml (hunk 1 above)
Comment (Kokan) [reply .travis.yml]: You are right, if it is a go I'll update this.

**Comment by czanik** [comment]
Hi,
I have to check it, but AFAIR, there is no RabbitMQ in CentOS/RHEL or EPEL (only in Fedora). And only the server, but no client in openSUSE (and SLES usually has a narrower selection of packages).

**Comment by czanik** [comment]
RHEL7 has librabbitmq 0.8.0 in the base repository.
RHEL6 has librabbitmq 0.5.2 in EPEL
SLES15 has no rabbitmq at all
openSUSE only has the rabbitmq server and Java client

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by mitzkia** [comment]
Could you please check and add librabbitmq-dev package into dbld's package lists.

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by lbudai** [comment]
OBS won't be a problem as I'm building the rabbitmq package inside OBS.

**Comment by lbudai** [comment]
@kira-syslogng retest this please

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**